app/src/shared/components/Modal/index.jsx

```diff
@@ -0,0 +1,30 @@
+// @flow
+
+import React, { type Node } from 'react'
+import { createPortal } from 'react-dom'
+
+import BodyClass, { NO_SCROLL } from '$shared/components/BodyClass'
+import Context from '$shared/contexts/Modal'
+import NoModalRootError from '$shared/errors/NoModalRootError'
+
+type Props = {
+    children: Node,
+}
+
+class Modal extends React.Component<Props> {
+    static contextType = Context
+
+    static isOpen() {
+        return BodyClass.includes(NO_SCROLL)
+    }
+
+    render() {
+        const { root } = this.context
+        if (!root) {
+            throw new NoModalRootError()
+        }
+        return createPortal(this.props.children, root)
```

Comment:
Hmm, Portals might not work when we are enabling SSR. Maybe we could use e.g. https://github.com/tajo/react-portal or https://github.com/cloudflare/react-gateway?